File: hw/sanchez/u3/hw27Observer/Observer/ObserverJs/model/Investor.js
const IInvestor = require("./IInvestor");

class Investor extends IInvestor {

    constructor(name) {
        super();
        this.name = name;
    }

    update(stock) {
        console.log(
            `Notified ${this.name} of ${stock.symbol} change to ${stock.price}`
        );
    }

    getName() {
        return this.name;
    }
}

module.exports = Investor;

File: hw/sanchez/u3/hw27Observer/Observer/ObserverPy/model/investor.py
from model.i_investor import IInvestor


class Investor(IInvestor):

    def __init__(self, name):
        self.name = name

    def update(self, stock):
        print(
            f"Notified {self.name} of "
            f"{stock.symbol} change to {stock.price}"
        )

    def get_name(self):
        return self.name